resotocore/resotocore/core_config.py

```diff
@@ -646,16 +667,26 @@ def parse_config(args: Namespace, core_config: Json, command_templates: Optional
         if value is not None:
             adjusted = set_value_in_path(value, path, adjusted)
 
-    # coerce the resulting json to the config model
-    try:
-        model = Model.from_kinds(from_js(config_model(), List[Kind]))
-        root = model.get(ResotoCoreRoot)
-        if isinstance(root, ComplexKind):
-            adjusted = root.coerce(adjusted)
-    except Exception as e:
-        log.warning(f"Can not adjust configuration: {e}", exc_info=e)
+    # here we only care about the resotocore overrides
+    core_config_overrides = (get_core_overrides() or {}).get(ResotoCoreRoot)
+    # merge the file overrides into the adjusted config
+    if core_config_overrides:
+        adjusted = merge_json_elements(adjusted, core_config_overrides)
 
     try:
+        # replacing the env vars and exploding if there is no value
+        adjusted = replace_env_vars(adjusted, os.environ, ignore_missing=False)
```

Comment:
The exploding part will be catched by the surrounding try/catch block.
The result will be the default configuration. I guess this error should bubble up?
Example:
```
23-02-28 08:33:21|ERROR|Final configuration can not be parsed! Fall back to default configuration.
ValueError: Env var not found: NOT_PROVIDED_ENV_VAR
```